File: itmd-521/labs/week-04/MnMCount.py
#Vinutha_Bengaluru_Prabhudev
import sys
from pyspark.sql import SparkSession

if __name__ =="_main_":
    if len(sys/argv) != 2:
        print("Usage mnmcount <file>", file=sys.stderr)
        sys.exit(-1)

    spark = (SparkSession.builder.appName.("PythonMnMcount").getOrCreate())
    mnm_file = sys.argv[1]
    mnm_df=(spark.read.format("csv")
            .option("header","true")
            .option("inferSchema","true")
            .load(mnm_file))

    count_mnm_df = (mnm_df. select("State","Color","Count")
    .groupBy("State","Color")
    .sum("Count")
    .orderBy("sum(Count)", ascending=False))

    count_mnm_df.show(n=60, truncate=False)
    print("Total Rows = %d " %(count_mnm_df.count()))

    ca_count_mnm_df = (mnm_df
        .select("State","Color","Count")
        .where(mnm_df.state=="CA")
        .groupBy("State","Color")
        .sum("Count")
        .orderBy("sum(Count)", ascending = False))


    ca_count-mnm_df.show(n=10, truncate=Falsespark)
    spark.stop()

File: itmd-521/labs/week-04/src/main/scala/chapter2/MnMcount.scala
package main.scala.chapter2

import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions._

object MnMcount {
  def main(args: Array[String]) {
    val spark = SparkSession
      .builder
      .appName("MnMCount")
      .getOrCreate()

    if (args.length < 1) {
      print("Usage: MnMcount <mnm_file_dataset>")
      sys.exit(1)
    }

    val mnmFile = args(0)
    
    val mnmDF = spark.read.format("csv")
      .option("header", "true")
      .option("inferSchema", "true")
      .load(mnmFile)
    
    mnmDF.show(5, false)

    
    val countMnMDF = mnmDF.select("State", "Color", "Count")
        .groupBy("State", "Color")
        .sum("Count")
        .orderBy(desc("sum(Count)"))

    
    countMnMDF.show(60)
    println(s"Total Rows = ${countMnMDF.count()}")
    println()

    
    val caCountMnNDF = mnmDF.select("*")
      .where(col("State") === "CA")
      .groupBy("State", "Color")
      .sum("Count")
      .orderBy(desc("sum(Count)"))


    caCountMnNDF.show(10)
    
    spark.stop()
  }
}
